nsxt/resource_nsxt_edge_transport_node.go

```diff
@@ -1147,8 +1173,36 @@ func getHostSwitchProfileResourceType(m interface{}, id string) (string, error)
 	return "", fmt.Errorf("Host Switch Profile type not found for %s", id)
 }
 
-func getHostSwitchProfileIDsFromSchema(m interface{}, hswProfileList []interface{}) ([]mpmodel.HostSwitchProfileTypeIdEntry, error) {
+func getHostSwitchProfileIDsFromSchema(m interface{}, parentMap map[string]interface{}) ([]mpmodel.HostSwitchProfileTypeIdEntry, error) {
 	var hswProfiles []mpmodel.HostSwitchProfileTypeIdEntry
+
+	uplinkProfile := parentMap["uplink_profile"].(string)
+	haProfile := parentMap["vtep_ha_profile"].(string)
+
+	if len(uplinkProfile) > 0 {
+		profileType := mpmodel.BaseHostSwitchProfile_RESOURCE_TYPE_UPLINKHOSTSWITCHPROFILE
+		elem := mpmodel.HostSwitchProfileTypeIdEntry{
+			Key:   &profileType,
+			Value: &uplinkProfile,
+		}
+		hswProfiles = append(hswProfiles, elem)
+	}
+
+	if len(haProfile) > 0 {
+		profileType := mpmodel.BaseHostSwitchProfile_RESOURCE_TYPE_VTEPHAHOSTSWITCHPROFILE
+		elem := mpmodel.HostSwitchProfileTypeIdEntry{
+			Key:   &profileType,
+			Value: &haProfile,
+		}
+		hswProfiles = append(hswProfiles, elem)
+	}
+
+	if len(hswProfiles) > 0 {
+		return hswProfiles, nil
```

Comment:
I think this return is tantamount to ignoring host_switch_profiles is the profile specific attributes were set.
In that case why not use conflictsWith?

Reply:
Answered above